src/lib/models/event-history.ts

```diff
@@ -1,24 +1,31 @@
+import { convertPayloadToJson } from '$lib/utilities/decode-payload';
 import type { GetWorkflowExecutionHistoryResponse, HistoryEvent } from '$types';
 
-export function getEventAttributes(historyEvent: HistoryEvent): EventAttribute {
+export async function getEventAttributes(
+  historyEvent: HistoryEvent,
+): Promise<EventAttribute> {
   const attributeKey = Object.keys(historyEvent).find((key) =>
     key.includes('Attributes'),
   );
+  const attributes = await convertPayloadToJson(historyEvent[attributeKey]);
+
   return {
     type: attributeKey as EventAttributeKeys,
-    ...historyEvent[attributeKey],
+    ...attributes,
   };
 }
 
-export const toEventHistory = (
+export const toEventHistory = async (
   response: GetWorkflowExecutionHistoryResponse,
-): HistoryEventWithId[] => {
-  return response.history.events.map(
-    (event): HistoryEventWithId => ({
-      ...event,
-      id: String(event.eventId),
-      eventType: event.eventType as unknown as EventType,
-      attributes: getEventAttributes(event),
-    }),
+): Promise<HistoryEventWithId[]> => {
+  return await Promise.all(
```

Comment:
My only fear with `Promise.all()` is that if _one_ of them fails, the entire thing fails. But, I think we can deal with that later. I know you don’t love observables, but we’re teetering on the edge eventually, but that’s an undertaking outside of the scope of this PR, so I think this  works for now and we can refine our thinking down the road.

Reply:
You probably don’t need the await since the consumer is going to need to resolve the promise anyway.

Reply:
I'm down with observables in the future.

Probably right on the await, I had done it because of difficulty with types and it just stuck and hadn't reviewed it. I had pushed this before checking first.